### ISSUE TITLE
There are prerequisites, we should ensure that they are installed.

### ISSUE THREAD

**Issue by cheilman** [cheilman/carapace#21]
There are a bunch of packages, etc. that need to be installed before we can install carapace.  At a minimum, we should check for these and bail out if they aren't there.  Better would be to install them.

For example (a lot of these may be standard):

- aptitude?
- bash
- coreutils?
- curl
- cut
- du
- git
- grep
- gzip/gunzip
- pv
- sed
- tar
- vim
- zsh (obvs)

Also these non-required packages are nice to have:

- golang
- imagemagick
- ncdu
- powerline
- silversearcher-ag
- tree
